src/index.ts

```diff
@@ -16,66 +17,95 @@ export interface Text extends BaseOptions {}
 
 // provide a image for loading
 export interface Image extends BaseOptions {
-  src: string
+  src: string;
 }
 
 // use svg loading animation for loading
 interface _Svg extends BaseOptions {
-  content?: string
-  path?: string
+  content?: string;
+  path?: string;
 }
-export type LoadingPlaceholderType = 'text' | 'img' | 'svg'
-export type Svg = _Svg & ({ content: string } | { path: string })
+export type LoadingPlaceholderType = "text" | "img" | "svg";
+export type Svg = _Svg & ({ content: string } | { path: string });
 
-export type Options = Text | Image | Svg
-export function spaLoading(type: 'text', options?: Text): Plugin
-export function spaLoading(type: 'img', options?: Image): Plugin
-export function spaLoading(type: 'svg', options?: Svg): Plugin
-export function spaLoading(type: LoadingPlaceholderType = 'text', options: any): Plugin {
+export type Options = Text | Image | Svg;
+export function spaLoading(type: "text", options?: Text): Plugin;
+export function spaLoading(type: "img", options?: Image): Plugin;
+export function spaLoading(type: "svg", options?: Svg): Plugin;
+export function spaLoading(
+  type: LoadingPlaceholderType = "text",
+  options: any
+): Plugin {
   const defaultOptions = {
-    tipText: 'Loading...',
-    rootElementId: 'app',
+    tipText: "Loading...",
+    rootElementId: "app",
     debounce: 150,
-    devEnable: true
-  }
-  options = Object.assign(defaultOptions, options)
-  let root = process.cwd()
-  let appNodeRegexp = new RegExp(`<div id="${options.rootElementId}">([\\w\\W]*)<\\/div>`, 'gim')
+    devEnable: true,
+    css: "",
+    cssPath: "",
+  };
+  options = Object.assign(defaultOptions, options);
+
+  let root = process.cwd();
+
+  let appNodeRegexp = new RegExp(
+    `<div id="${options.rootElementId}">([\\w\\W]*)<\\/div>`,
+    "gim"
+  );
   const aniMap: Record<LoadingPlaceholderType, any> = {
     img: (config: Image) => `<img src="${config.src}" alt="loading img">`,
     svg: (config: Svg) => config.content,
-    text: () => ''
-  }
-  const renderTemplate = (config: Options) => {
-    const halfDebounce = config.debounce! / 2
+    text: () => "",
+  };
+
+  const renderTemplate = (config: Options, externalStyle: string = "") => {
+    const halfDebounce = config.debounce! / 2;
     return `
+    ${
```

Comment:
`external-css` should below `internal-css`, otherwise can not insufficient weight to override internal css。and style tag id attr better use `user-css`  markup it's user-specified css

Reply:
OK，I think external css weight should be lower than internal css. Actually, I think most of people should't like to write css string. So internal css overwrite should no matter.

Reply:
In this case, if the user wants to customise a certain style, he needs to increase the weight of the style by himself. Isn't it a little inconvenient to add an important?

Reply:
Yes, sorry, I misunderstood what you meant. 
Wait to change code to increase the weight.